doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -689,198 +734,548 @@ one from Ltac1, except that it requires the goal to be focused.
-this features has the same semantics as in Ltac1. In particular, a ``reverse``
-flag can be specified to match hypotheses from the more recently introduced to
-the least recently introduced one.
+   - ``constr`` for pattern variables appearing in a :n:`@term`
+   - ``Pattern.context`` for variables binding a context
+   - ``ident`` for variables binding a hypothesis name.
+
+   The same identifier caveat as in the case of matching over constr applies, and
+   this features has the same semantics as in Ltac1. In particular, a ``reverse``
+   flag can be specified to match hypotheses from the more recently introduced to
+   the least recently introduced one.
 
 .. _ltac2_notations:
 
 Notations
 ---------
 
-Notations are the crux of the usability of Ltac1. We should be able to recover
-a feeling similar to the old implementation by using and abusing notations.
+.. cmd:: Ltac2 Notation @string {* @syn_node ( @ltac2_scope ) } {? : @int } := @tac2expr6
+   :name: Ltac2 Notation
 
-Scopes
-~~~~~~
+   .. todo JEF explain components, e.g. string and int
 
-A scope is a name given to a grammar entry used to produce some Ltac2 expression
-at parsing time. Scopes are described using a form of S-expression.
+   .. todo seems like syn_node maybe should use lident rather than ident, considering:
 
-.. prodn::
-   ltac2_scope ::= {| @string | @int | @lident ({+, @ltac2_scope}) }
+      Ltac2 Notation "ex1" X(constr) := print (of_constr X).
+      ex1 1.
 
-A few scopes contain antiquotation features. For the sake of uniformity, all
-antiquotations are introduced by the syntax :n:`$@lident`.
+      Unbound constructor X
 
-The following scopes are built-in.
+      This works find with lower-case "x" in place of "X"
 
-- :n:`constr`:
+   .. todo Ltac2 Notation := permits redefining same symbol (no warning)
+      Also allows defining a symbol beginning with uppercase, which is prohibited
+      in similar constructs.
 
-  + parses :n:`c = @term` and produces :n:`constr:(c)`
+   :cmd:`Ltac2 Notation` provides a way to extend |Coq|'s syntax.  The left-hand
+   side (before the `:=`) defines the syntax to recognize and gives formal parameter
+   names for the syntactic values.  When the notation is used, the values are substituted
+   into the right-hand side.  In the following example, `x` is the formal parameter name, `constr` is
+   the name of a nonterminal describing the syntax (which the documentation calls :token:`term`).
+   `print` and `of_constr` are functions provided by |Coq| through `Message.v`.
 
-  This scope can be parameterized by a list of delimiting keys of notation
-  scopes (as described in :ref:`LocalInterpretationRulesForNotations`),
-  describing how to interpret the parsed term. For instance, :n:`constr(A, B)`
-  parses :n:`c = @term` and produces :n:`constr:(c%A%B)`.
+   .. todo "print" doesn't seem to pay attention to "Set Printing All"
 
-- :n:`ident`:
+   EXAMPLE FAILS IN COQTOP DIRECTIVE, WORKS IN coqtop/CoqIDE::
```

Comment:
Any way these examples could be done without ever calling `reset` in all the Ltac2 chapter?

Reply:
Yeah, I figured that out the other day.  (I had not understood that `abort` was applied after the following block of Coq code.